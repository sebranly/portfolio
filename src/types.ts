// TODO: have a glossary on the website
// TODO: explain myProjects series
export enum Tag {
  All = 'All',
  ThreeDimensional = '3D',
  AIML = 'AIML',
  Archived = 'archived',
  C = 'C',
  CSS = 'CSS',
  CLI = 'CLI',
  DNS = 'DNS',
  GitHub = 'GitHub',
  GUI = 'GUI',
  HTML = 'HTML',
  i18n = 'i18n',
  JavaScript = 'JavaScript',
  Live = 'live',
  MyProjects = 'myProjects',
  NintendoDS = 'Nintendo DS',
  OpenGL = 'OpenGL',
  Puzzle = 'puzzle',
  Python = 'Python',
  React = 'React',
  SchoolProject = 'school project',
  SDL = 'SDL',
  VideoGame = 'video game',
  Website = 'website',
  PAlib = 'PAlib',
  PHP = 'PHP',
  SQL = 'SQL',
  TailwindCSS = 'TailwindCSS',
  TypeScript = 'TypeScript',
  Windows = 'Windows'
}

export enum Color {
  Black = 'black',
  White = 'white',
  Gray = 'gray',
  Red = 'red',
  Orange = 'orange',
  Amber = 'amber',
  Yellow = 'yellow',
  Green = 'green',
  Blue = 'blue',
  Indigo = 'indigo',
  Purple = 'purple',
  Pink = 'pink',
  Rose = 'rose'
}

export enum Role {
  Algorithm = 'Algorithm',
  Developer = 'Developer',
  Designer = 'Designer',
  ProductOwner = 'Product Owner',
  Tester = 'Tester'
}

export type Contributor = {
  female?: boolean;
  name: string;
  roles: Role[];
};

export type Project = {
  download?: string;
  title: string;
  contributors?: Contributor[];
  github?: string;
  subtitle: string;
  description: string[];
  tags: Tag[];
  video?: string;
  website?: string;
  years: number[];
  textualSnapshot?: string[];
};

export type Language = {
  shortName: string;
  name: string;
};
