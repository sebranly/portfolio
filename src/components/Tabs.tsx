import * as React from 'react';
import classnames from 'classnames';
import { Tag } from '../types';
import { TAB_COUNT } from '../constants';
import { useTranslation } from 'react-i18next';

export interface TabsProps {
  applicationType: Tag;
  onClickTab: (tag: Tag) => void;
}

const Tabs = React.forwardRef((props: TabsProps, ref) => {
  const { applicationType, onClickTab } = props;
  const { t } = useTranslation();

  return (
    <div ref={ref as any} className="mt-4 flex w-full">
      {[Tag.All, Tag.CLI, Tag.GUI, Tag.Website].map((tag: Tag, index: number) => {
        const text = t(`selector.${tag}`);
        const isSelected = applicationType === tag;
        const classnamesTab = classnames(
          {
            'bg-black': isSelected,
            'text-white': isSelected,
            'rounded-tl-lg': isSelected && index !== 0,
            'rounded-tr-lg': isSelected && index !== TAB_COUNT - 1
          },
          'p-2 flex-1'
        );

        return (
          <button
            className={classnamesTab}
            key={tag}
            style={{ fontWeight: isSelected ? 'bold' : 'normal' }}
            type="submit"
            onClick={() => onClickTab(tag)}
          >
            {text}
          </button>
        );
      })}
    </div>
  );
});

export { Tabs };
