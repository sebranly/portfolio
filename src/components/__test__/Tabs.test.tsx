import * as React from 'react';
import { render } from '@testing-library/react';

import { Tabs } from '../Tabs';
import { Tag } from '../../types';

test('Tabs', () => {
  const createComponent = () => <Tabs applicationType={Tag.CLI} onClickTab={jest.fn()} />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
