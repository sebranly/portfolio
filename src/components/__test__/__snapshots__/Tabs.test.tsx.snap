// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs 1`] = `
NodeList [
  <div
    class="mt-4 flex w-full"
  >
    <button
      class="p-2 flex-1"
      style="font-weight: normal;"
      type="submit"
    >
      selector.All
    </button>
    <button
      class="bg-black text-white rounded-tl-lg rounded-tr-lg p-2 flex-1"
      style="font-weight: bold;"
      type="submit"
    >
      selector.CLI
    </button>
    <button
      class="p-2 flex-1"
      style="font-weight: normal;"
      type="submit"
    >
      selector.GUI
    </button>
    <button
      class="p-2 flex-1"
      style="font-weight: normal;"
      type="submit"
    >
      selector.website
    </button>
  </div>,
]
`;
